src/sardana/macroserver/scan/gscan.py

```diff
@@ -369,6 +371,43 @@ def __init__(self, macro, generator=None, moveables=[], env={},
         # ---------------------------------------------------------------------
         self._setupEnvironment(env)
 
+    def _check_moveables_limits(self):
+        for m in self._moveables:
+            pos_range = m.moveable.getAttribute("Position").range
+            try:
+                if type(pos_range[1]) == str:
```

Comment:
I think that `isinstance` is better than type comparison. But the best would be simply "duck typing" . The same to the rest of comparisons.